### PR TITLE
Add _conf variants of SRS intrinsics

### PR DESCRIPTION
**Pull request by konstantinschwarz**

### CODE REVIEW

File: clang/test/CodeGen/aie/aie2/aiev2-srs-intrinsics.cpp [hunk 1]
```diff
@@ -15,115 +15,469 @@
...
 // CHECK-NEXT:  entry:
-// CHECK-NEXT:    [[TMP0:%.*]] = tail call noundef <16 x i16> @llvm.aie2.I256.v16.acc64.srs(<16 x i64> [[ACC:%.*]], i32 [[SHFT:%.*]], i32 [[SIGN:%.*]])
-// CHECK-NEXT:    ret <16 x i16> [[TMP0]]
+// CHECK-NEXT:    [[TMP0:%.*]] = tail call noundef <32 x i8> @llvm.aie2.I256.v32.acc32.srs(<16 x i64> [[ACC:%.*]], i32 [[SHFT:%.*]], i32 [[SIGN:%.*]])
+// CHECK-NEXT:    ret <32 x i8> [[TMP0]]
 //
-v16int16 test_ssrs_gen1(v16acc64 acc, int shft, int sign) {
-   return ssrs(acc,shft,sign);
+v32uint8 test_ussrs(v32acc32 acc, int shft, int sign) {
+  return ussrs(acc, shft, sign);
```
Comment (martien-de-jong) [inline clang/test/CodeGen/aie/aie2/aiev2-srs-intrinsics.cpp:52]: Don't you think one union of socialist sovjet republic suffices? X:D

File: clang/lib/Headers/aiev2_srs.h [hunk 2]
```diff
@@ -16,199 +16,637 @@ INTRINSIC(v32int8) ssrs(v32acc32 acc, int shft, int sign) {
   return __builtin_aiev2_I256_v32_acc32_srs(acc, shft, sign);
 }
 
+INTRINSIC(v32int8)
+ssrs_conf(v32acc32 acc, int shft, int sign, crsat_t sat, crrnd_t rnd) {
+  crsat_t prev_sat = get_sat();
+  crrnd_t prev_rnd = get_rnd();
+  set_satmode(sat);
+  set_rnd(rnd);
+  v32int8 val = __builtin_aiev2_I256_v32_acc32_srs(acc, shft, sign);
```
Comment (gbossu) [inline clang/lib/Headers/aiev2_srs.h:25]: Super-nit: I'd have a slight preference for calling `ssrs(acc, shft, sign)` instead of our builtin. But at this point, I think it's just personal preference. Ignore me :)

**Review by gbossu — APPROVED** [review]
LGTM, thanks for harmonizing the tests and intrinsic definitions.

File: clang/test/CodeGen/aie/aie2/aiev2-srs-intrinsics.cpp [hunk 3]
```diff
@@ -15,115 +15,469 @@
...
+  return ussrs(acc, shft);
+}
+
+// CHECK-LABEL: @_Z9test_lsrsDv16_u7__acc64i(
 // CHECK-NEXT:  entry:
 // CHECK-NEXT:    [[TMP0:%.*]] = tail call noundef <16 x i32> @llvm.aie2.I512.v16.acc64.srs(<16 x i64> [[ACC:%.*]], i32 [[SHFT:%.*]], i32 1)
 // CHECK-NEXT:    ret <16 x i32> [[TMP0]]
 //
-v16int32 test_lsrs_signed0(v16acc64 acc, int shft) {
-   return lsrs(acc,shft,1);
```
Comment (martien-de-jong) [inline clang/test/CodeGen/aie/aie2/aiev2-srs-intrinsics.cpp:109]: dummy question: why this change?

File: clang/test/CodeGen/aie/aie2/aiev2-srs-intrinsics.cpp [hunk 4]
```diff
@@ -15,115 +15,469 @@
...
-v16bfloat16 test_to_v16bfloat16(v16accfloat acc){
-    return to_v16bfloat16(acc);
+v16bfloat16 test_to_v16bfloat16(v16accfloat acc) {
+  return to_v16bfloat16(acc);
 }
 
-// CHECK-LABEL: @_Z14test_ssrs_gen0Dv32_u7__acc32ii(
+// CHECK-LABEL: @_Z9test_ssrsDv32_u7__acc32ii(
 // CHECK-NEXT:  entry:
 // CHECK-NEXT:    [[TMP0:%.*]] = tail call noundef <32 x i8> @llvm.aie2.I256.v32.acc32.srs(<16 x i64> [[ACC:%.*]], i32 [[SHFT:%.*]], i32 [[SIGN:%.*]])
```
Comment (martien-de-jong) [inline clang/test/CodeGen/aie/aie2/aiev2-srs-intrinsics.cpp:24]: nit: Do we want to keep the flat I256 next to the structured v32.acc32 in the llvm intrinsic name? I find it untidy.

File: clang/lib/Headers/aiev2_srs.h (hunk 2 above)
Comment (SagarMaheshwari99) [reply clang/lib/Headers/aiev2_srs.h:25]: > Super-nit: I'd have a slight preference for calling ssrs(acc, shft, sign) instead of our builtin. But at this point, I think it's just personal preference. Ignore me :)

+1
